### PR TITLE
Prepare for clarinet integrate

### LINKED ISSUE #571: [stacking-pool] Make stacking pool work on devnet with pox-3
**Is your feature request related to a problem? Please describe.**
Integration tests can be done with `clarinet integrate` and pox-3.

**Describe the solution you'd like**
Adapt the code so that integration tests with the sbtc-signer ui can work together.

This issue is about collection changes required to make this happen. Similar changes would be needed for testnet as well.

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds a script to run clarinet integrate with ST000..pox-3 instead of .pox-3
* fixes #571 

builds on top of #619

### CODE REVIEW

**Comment by kantai** [comment]
@friedger -- would you be interested in a clarinet version that applied the PoX locks in normal clarinet tests (rather than integrate)? This PR achieves that (https://github.com/hirosystems/clarinet/pull/1074), but it currently requires building clarinet from that PR.

**Comment by friedger** [comment]
@kantai Nice! Currently, we use a local mocking pox-3 contract.

**Comment by MarvinJanssen** [comment]
Your script modifies the original source file right?

**Comment by friedger** [comment]
@MarvinJanssen correct

**Comment by MarvinJanssen** [comment]
Is that desirable? Should we make an integrate environment much like how we deal with tests?

**Comment by friedger** [comment]
@MarvinJanssen This was the fastest solution for now. I am not worried because it is a limited use case.

Eventually, I want a setup where deploy the contracts and launch the UI.

**Comment by MarvinJanssen** [comment]
We can add a few lines to copy all contracts to a dot directory and perform the replace there. Otherwise we should just check the changes into the source files.

**Comment by friedger** [comment]
@MarvinJanssen I have added .integration folder that is used to launch devnet

**Review by MarvinJanssen — APPROVED** [review]
Noticed you merged the new error code format as well.

LGTM.

**Comment by friedger** [comment]
Better to merge into #619  branch?

**Comment by MarvinJanssen** [comment]
I closed #619 because all changes were merged into #605. We can merge this one as-is I think. (Also looking at the title of the PR.)

File: sbtc-mini/contracts/sbtc-stacking-pool.clar [hunk 1]
```diff
@@ -342,12 +342,14 @@
         (asserts! (is-eq (get-current-window) registration)  err-not-in-registration-window)
 
         ;; Delegate-stx to their PoX address
-        (unwrap! (contract-call? .pox-3 delegate-stx amount-ustx (as-contract tx-sender) none (some pox-addr)) err-pre-registration-delegate-stx)
+        (match (contract-call? .pox-3 delegate-stx amount-ustx (as-contract tx-sender) none (some pox-addr))
+            success true
+            error (asserts! false (err (+ (to-uint error) u60000))))
```
Comment (setzeus) [inline sbtc-mini/contracts/sbtc-stacking-pool.clar:348]: Thank you for this. It's clear reading how throwing the returned error is much more specific than a single error message from the calling function.

(nothing to change/resolve)

**Review by setzeus — APPROVED** [review]
Reviewed all changes done to sbtc-peg-transfer & sbtc-stacking-pool. Can confirm majority of changes were made to either adhere to our[ new team guidelines](https://docs.google.com/document/d/1gtPdmuBDgmmeqPucez9dxpPDHK2A-rUesEp6zxaHtVU/edit) or to improve error messages from functions that contract hop.

LGTM.

**Pull request merged**